src/main/scala/apps/nearestNeighbour.scala

```diff
@@ -13,6 +14,13 @@ object nearestNeighbour {
     (f32 x f32) ->: f32 ->: f32 ->: f32
   )
 
+  // FIXME: could not find original Lift expression, this is made up
+  val highLevel: ToBeTyped[Expr] = depFun((n: Nat) => fun(
+    (n`.`(f32 x f32)) ->: f32 ->: f32 ->: (n`.`f32)
+  )((locations, lat, lng) =>
+    locations |> map(fun(loc => distance(loc)(lat)(lng)))
+  ))
+
   val nn: ToBeTyped[Expr] = depFun((n: Nat) => fun(
```

Comment:
For (some) of the other apps you went for `nameHL` and `nameLL`, here you have `highLevel` vs. `name`.
Could we have a unified naming scheme?

Reply:
Yes we can do that, do you have a preference between the two? I think that `highLevel` can be too vague sometimes (i.e. if we have multiple highLevel expression like for mriQ benchmarks). We could also go for a combination of both like `nameHighLevel` and `nameLowLevel` (or `nameNvidia`, basically any implementation qualifier).

Reply:
`nameHighLevel` vs `nameLowLevel` sound good to me. The name could sometimes include a version, so `nameVersionHighLevel` (as in `mmNvidiaHighLevel`).

Reply:
I tried to make the names more consistent, let me know if that is good enough.